di2f/index.md

```diff
@@ -0,0 +1,226 @@
+
+## Threads
+
+A thread is topic-based collection of single-writer logs. A collection of logs represent updates to the "state" of an object (or dataset). The basic units of a thread -- logs and records -- provide a framework for creating, storing, and transmiting data in a p2p distributed network. The thread protocol is outline in detail in the threads whitepaper [@pick2020protocol].
+
+A log within a thread is essentially a set of cryptographically linked (i.e., hash-linked) records, that form a specific type of Merkle-DAG (directed acyclic graph) that represents a purely functional [@okasaki1999purely] and authenticated [@tamassia2003authenticated] (i.e., immutable) singly-linked list. The key insight here is that, assuming two peers have received all of the same updates to a thread, they will deterministically arrive at the same thread structure, and that thread structure can be summarized by the (set of) hash(es) of the head(s) of the underlying log(s).
+
+# Thread DIDs
+
+<!-- @sanderpick What are we trying to "solve" or say in this section specifically? Why, in one or two sentences, do we need to build this? -->
+
+In practice, a thread _network_ may have multiple actors and structures (subjects) that can be described by DID documents. Subjects include any of the following entities:
+
+Peers
+: _Peers_ (e.g., `did:p2p:foo`) can offer network services, such as thread "hosting", pinning services (i.e., IPFS, IPNS), Filecoin anchoring, database abstractions, API services, and more. A thread peer's DID is derived from its embedded networking host's key (which in practice is a `libp2p` ([`https://libp2p.io/`](https://libp2p.io/)) peer]).
+
+Users
+: A _User_ (e.g., `did:key:foo`, `did:3:foo`, `did:ethr:foo`, etc.) is any external identity that represents a network user, and that may interact with the network _via_ a Peer. These may be identified by any verifiable DID.
+
+Thread:
+: _Thread DID_ (e.g., `did:thread:id`) documents contain verification methods for all valid log signers. Other thread info such as the _log head_, _log metadata_, and _thread encryption keys_ are _not_ stored in the DID document, as this information is only needed by peers that are sharing a thread, and can be more efficiently exchanged using the thread protocol [@pick2020protocol] directly (vs. a global document registry). _Log addresses_ are referenced as _service endpoints_, as defined in [@reed2020decentralized] sec 5.4.
+
+<!-- @sanderpick Rather than introducing a new term here (log signers), could we just say for all valid peers and/or users? -->
```

Comment:
@sanderpick Rather than introducing a new term here (log signers), could we just say for all valid peers and/or users?

Reply:
Yep, sgtm